core/src/main/scala/akka/kafka/internal/DeferredProducer.scala

```diff
@@ -57,7 +59,7 @@ private[kafka] trait DeferredProducer[K, V] {
   }
 
   final protected def resolveProducer(settings: ProducerSettings[K, V]): Unit = {
-    val producerFuture = settings.createKafkaProducerAsync()(materializer.executionContext)
+    val producerFuture = settings.createKafkaProducerAsync()(interpreter.materializer.executionContext)
```

Comment:
Because `materializer` is protected, I presume.

That is not great since `interpreter` is `private[akka]` internal API. Yes, it can be accessed from this akka package but we want to avoid using internal apis like that.

Would it be possible to change `trait DeferredProducer` to 
```
abstract class DeferredProducer extends GraphStageLogic with StageIdLogging
```

Reply:
Seems to work 👍